README.md

```diff
@@ -45,6 +58,33 @@ test.RunE2e(t, true /*runFullTest*/)
 test.RunSdk(t, "nodejs" /*language*/)
 ```
 
+### Upgrade Tests
+
+Set these extra options to enable upgrade tests:
+
+```go
+func TestSimple(t *testing.T) {
+  test := NewProviderTest("test/simple",
+    WithProviderName("gcp"),
+    WithBaselineVersion("6.67.0"),
```

Comment:
Far out of scope of this PR: In a future version I would love if we could do this without naming a baseline version.

Ideally, we want to check (at least quickest) that version vX.0.0 to vX.Y.Z work with vX.Y.Z as the release candidate.

Reply:
Perhaps a ticket here? I think we do need to be very explicit about the baseline version when recording. We can pick latest recorded when replaying. At some point we might need a nightly workflow bumps the baseline version and its recordings semi-automatically. A related question we were pondering with Daniel is whether recordings should live in repo or in a sidecar location for repo-specific data...

There's lots of open questions I filed https://github.com/pulumi/providertest/issues/18 for now.